stats/stats.go

```diff
@@ -36,19 +36,22 @@ type RPCStats interface {
 	IsClient() bool
 }
 
-// Begin contains stats when an RPC begins.
+// Begin contains stats when an RPC attempt begins.
 // FailFast is only valid if this Begin is from client side.
 type Begin struct {
 	// Client is true if this Begin is from client side.
 	Client bool
-	// BeginTime is the time when the RPC begins.
+	// BeginTime is the time when the RPC attempt begins.
 	BeginTime time.Time
 	// FailFast indicates if this RPC is failfast.
 	FailFast bool
 	// IsClientStream indicates whether the RPC is a client streaming RPC.
 	IsClientStream bool
 	// IsServerStream indicates whether the RPC is a server streaming RPC.
 	IsServerStream bool
+	// IsTransparentRetryAttempt indicates whether this attempt was initiated
+	// due to transparently retrying a previous attempt.
+	IsTransparentRetryAttempt bool
```

Comment:
What's the purpose of this bool?

It won't know which attempt was the previous. This is just so the handler can ignore the stats? But if it can, it should ignore the previous instead, right?

Reply:
Only with hedging would it be potentially unclear which attempt was the previous attempt, but the point is not to relate to a specific prior attempt anyway.  This is here for metrics, as described here: https://github.com/grpc/proposal/blob/master/A45-retry-stats.md#measures